consensus/istanbul/backend/announce.go

```diff
@@ -589,130 +649,311 @@ func (sb *Backend) regossipAnnounce(msg *istanbul.Message, payload []byte) error
 	return nil
 }
 
-// ===============================================================
-//
-// define the AnnounceVersion messge format, the getAnnounceVersions send and handle function, and the announceVersions handle function
+// signedAnnounceVersion is a signed message from a validator indicating the most
+// recent version of its enode.
+type signedAnnounceVersion struct {
+	Address   common.Address
+	Version   uint
+	Signature []byte
+}
+
+func newSignedAnnounceVersionFromEntry(entry *vet.SignedAnnounceVersionEntry) *signedAnnounceVersion {
+	return &signedAnnounceVersion{
+		Address: entry.Address,
+		Version: entry.Version,
+		Signature: entry.Signature,
+	}
+}
+
+func (sav *signedAnnounceVersion) Sign(signingFn func(data []byte) ([]byte, error)) error {
+	payloadNoSig, err := sav.payloadNoSig()
+	if err != nil {
+		return err
+	}
+	sav.Signature, err = signingFn(payloadNoSig)
+	if err != nil {
+		return err
+	}
+	return nil
+}
 
-type announceVersion struct {
-	ValAddress         common.Address
-	AnnounceMsgVersion uint
+func (sav *signedAnnounceVersion) ECDSAPublicKey() (*ecdsa.PublicKey, error) {
```

Comment:
If you keep the `Address` field (see my previous comment on this PR), it may be useful to verify the address matches the recovered public key (returning an error if it does not)